NNPTPZ1/Program.cs

```diff
@@ -23,311 +10,159 @@ namespace NNPTPZ1
-                    Cplx x = obj as Cplx;
-                    return x.Re == Re && x.Imaginari == Imaginari;
+                    known = true;
+                    id = i;
                 }
-                return base.Equals(obj);
             }
-
-            public readonly static Cplx Zero = new Cplx()
-            {
-                Re = 0,
-                Imaginari = 0
-            };
-
-            public Cplx Multiply(Cplx b)
+            if (!known)
             {
-                Cplx a = this;
-                // aRe*bRe + aRe*bIm*i + aIm*bRe*i + aIm*bIm*i*i
-                return new Cplx()
-                {
-                    Re = a.Re * b.Re - a.Imaginari * b.Imaginari,
-                    Imaginari = (float)(a.Re * b.Imaginari + a.Imaginari * b.Re)
-                };
-            }
-            public double GetAbS()
-            {
-                return Math.Sqrt( Re * Re + Imaginari * Imaginari);
+                roots.Add(coordinates);
+                id = roots.Count;
+                maxid = id + 1;
             }
 
-            public Cplx Add(Cplx b)
-            {
-                Cplx a = this;
-                return new Cplx()
-                {
-                    Re = a.Re + b.Re,
-                    Imaginari = a.Imaginari + b.Imaginari
-                };
-            }
-            public double GetAngleInDegrees()
-            {
-                return Math.Atan(Imaginari / Re);
-            }
-            public Cplx Subtract(Cplx b)
-            {
-                Cplx a = this;
-                return new Cplx()
-                {
-                    Re = a.Re - b.Re,
-                    Imaginari = a.Imaginari - b.Imaginari
-                };
-            }
+            return id;
+        }
 
-            public override string ToString()
-            {
-                return $"({Re} + {Imaginari}i)";
-            }
+        private static Color CreatePixelColor(float it, int id)
+        {
+            Color selectedColor = colors[id % colors.Length];
+            Color adjustedColor = Color.FromArgb(selectedColor.R, selectedColor.G, selectedColor.B);
+            adjustedColor = Color.FromArgb(Math.Min(
+                Math.Max(0, selectedColor.R - (int)it * 2), 255), Math.Min(Math.Max(0, selectedColor.G - (int)it * 2), 255), Math.Min(Math.Max(0, selectedColor.B - (int)it * 2), 255));
+            return adjustedColor;
+        }
 
-            internal Cplx Divide(Cplx b)
-            {
-                // (aRe + aIm*i) / (bRe + bIm*i)
-                // ((aRe + aIm*i) * (bRe - bIm*i)) / ((bRe + bIm*i) * (bRe - bIm*i))
-                //  bRe*bRe - bIm*bIm*i*i
-                var tmp = this.Multiply(new Cplx() { Re = b.Re, Imaginari = -b.Imaginari });
-                var tmp2 = b.Re * b.Re + b.Imaginari * b.Imaginari;
+        private static void PaintPixel(Bitmap bitmap, int i, int j, Color selectedColor)
+        {
+            bitmap.SetPixel(j, i, selectedColor);
```

Comment:
you seriously messed up whole algorithm - now you use J for the witdth dimension, but before you used it with height in for loop. You need to recheck and fix every match between i,j - witdth,height dimensions